environs/bootstrap/config.go

```diff
@@ -288,6 +305,35 @@ func NewConfig(attrs map[string]interface{}) (Config, error) {
 		config.CAPrivateKey = caKeyPem
 	}
 
+	// Try get the key from the attribute map, if it isn't present,
+	// try load it from the default location.
+	if sshServerHostKey, ok := attrs[SSHServerHostKeyKey].(string); ok {
+		config.SSHServerHostKey = sshServerHostKey
+	} else {
+		var userSpecified bool
+		var err error
+		config.SSHServerHostKey, userSpecified, err = readFileAttr(attrs, SSHServerHostKeyKey, strings.ReplaceAll(SSHServerHostKeyKey, "-", "_"))
```

Comment:
This currently diverges from the standard i.e. both pem (public and private) files are expected to have `-` as separators, so let's keep things consistent.